In [1]:
!python -m pip install --upgrade pip setuptools wheel

  Obtaining dependency information for pip from https://files.pythonhosted.org/packages/b7/3f/945ef7ab14dc4f9d7f40288d2df998d1837ee0888ec3659c813487572faa/pip-25.2-py3-none-any.whl.metadata
  Obtaining dependency information for wheel from https://files.pythonhosted.org/packages/0b/2c/87f3254fd8ffd29e4c02732eee68a83a1d3c346ae39bc6822dcbcb697f2b/wheel-0.45.1-py3-none-any.whl.metadata
   ---------------------------------------- 0.0/1.8 MB ? eta -:--:--
   ---------------------------------------- 0.0/1.8 MB ? eta -:--:--
    --------------------------------------- 0.0/1.8 MB 445.2 kB/s eta 0:00:04
   -- ------------------------------------- 0.1/1.8 MB 1.2 MB/s eta 0:00:02
   ----- ---------------------------------- 0.2/1.8 MB 1.7 MB/s eta 0:00:01
   -------- ------------------------------- 0.4/1.8 MB 2.1 MB/s eta 0:00:01
   -------------- ------------------------- 0.6/1.8 MB 3.1 MB/s eta 0:00:01
   ---------------------- ----------------- 1.0/1.8 MB 4.1 MB/s eta 0:00:01
   ---------------

In [2]:
!pip install pandas seaborn matplotlib opencv-python python-Levenshtein imagehash scikit-learn pillow

  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
   ---------------------------------------- 0.0/11.0 MB ? eta -:--:--
   ---------------------------------------- 0.0/11.0 MB ? eta -:--:--
    --------------------------------------- 0.3/11.0 MB ? eta -:--:--
   -- ------------------------------------- 0.8/11.0 MB 1.4 MB/s eta 0:00:08
   --- ------------------------------------ 1.0/11.0 MB 1.6 MB/s eta 0:00:07
   ----- ---------------------------------- 1.6/11.0 MB 1.8 MB/s eta 0:00:06
   -------- ------------------------------- 2.4/11.0 MB 2.1 MB/s eta 0:00:05
   ----------- ---------------------------- 3.1/11.0 MB 2.4 MB/s eta 0:00:04
   --------------- ------------------------ 4.2/11.0 MB 2.7 MB/s eta 0:00:03
   ------------------- -------------------- 5.2/11.0 MB 3.1 MB/s eta 0:00:02
   ---------------------- ----------------- 6.3/11.0 MB 3.3 MB/s eta 0:00:02
   -------------------------- ------------- 7.3/11.0 MB 3.5 MB/s eta 0:0

  DEPRECATION: Building 'antlr4-python3-runtime' using the legacy setup.py bdist_wheel mechanism, which will be removed in a future version. pip 25.3 will enforce this behaviour change. A possible replacement is to use the standardized build interface by setting the `--use-pep517` option, (possibly combined with `--no-build-isolation`), or adding a `pyproject.toml` file to the source tree of 'antlr4-python3-runtime'. Discussion can be found at https://github.com/pypa/pip/issues/6334


In [3]:
!pip install tensorflow

   ---------------------------------------- 0.0/331.9 MB ? eta -:--:--
   ---------------------------------------- 0.5/331.9 MB 4.2 MB/s eta 0:01:19
   ---------------------------------------- 1.3/331.9 MB 4.2 MB/s eta 0:01:19
   ---------------------------------------- 2.4/331.9 MB 4.6 MB/s eta 0:01:12
   ---------------------------------------- 3.7/331.9 MB 5.0 MB/s eta 0:01:07
    --------------------------------------- 4.7/331.9 MB 4.9 MB/s eta 0:01:07
    --------------------------------------- 5.5/331.9 MB 4.8 MB/s eta 0:01:09
    --------------------------------------- 6.3/331.9 MB 4.6 MB/s eta 0:01:11
    --------------------------------------- 7.1/331.9 MB 4.5 MB/s eta 0:01:12
    --------------------------------------- 7.9/331.9 MB 4.5 MB/s eta 0:01:12
   - -------------------------------------- 8.9/331.9 MB 4.5 MB/s eta 0:01:13
   - -------------------------------------- 9.7/331.9 MB 4.4 MB/s eta 0:01:14
   - -------------------------------------- 10.5/331.9 MB 4.4 MB/s eta 

In [7]:
!pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu118

Looking in indexes: https://download.pytorch.org/whl/cu118
   ---------------------------------------- 0.0/5.5 MB ? eta -:--:--
   ------- -------------------------------- 1.0/5.5 MB 7.2 MB/s eta 0:00:01
   ----------------- ---------------------- 2.4/5.5 MB 6.7 MB/s eta 0:00:01
   ------------------- -------------------- 2.6/5.5 MB 6.3 MB/s eta 0:00:01
   ---------------------------- ----------- 3.9/5.5 MB 5.2 MB/s eta 0:00:01
   -------------------------------- ------- 4.5/5.5 MB 5.1 MB/s eta 0:00:01
   ---------------------------------------- 5.5/5.5 MB 4.7 MB/s  0:00:01
   ---------------------------------------- 0.0/2.8 GB ? eta -:--:--
   ---------------------------------------- 0.0/2.8 GB 5.6 MB/s eta 0:08:22
   ---------------------------------------- 0.0/2.8 GB 4.9 MB/s eta 0:09:30
   ---------------------------------------- 0.0/2.8 GB 4.5 MB/s eta 0:10:21
   ---------------------------------------- 0.0/2.8 GB 4.5 MB/s eta 0:10:21
   ---------------------------------------- 0.

  You can safely remove it manually.
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
executorch 1.0.0 requires torch<2.10.0,>=2.9.0, but you have torch 2.7.1+cu118 which is incompatible.


In [10]:
import torch
print(torch.__version__)  
print(torch.version.cuda)
torch.cuda.is_available()

2.9.0+cpu
None


False

In [11]:
!nvidia-smi

Thu Oct 23 22:26:08 2025       
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 581.15                 Driver Version: 581.15         CUDA Version: 13.0     |
+-----------------------------------------+------------------------+----------------------+
| GPU  Name                  Driver-Model | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GeForce RTX 3080 ...  WDDM  |   00000000:01:00.0  On |                  N/A |
| N/A   63C    P8             12W /  165W |     277MiB /  16384MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+-----

### Restart system

In [1]:
import torch
print(torch.__version__)  
print(torch.version.cuda)
torch.cuda.is_available()

2.9.0+cpu
None


False

# To use executorch

this overrides torch-2.7.1+cu118 with torch-2.9.0 and cant find cuda on nvidia geforce rtx 3080

In [1]:
!pip install executorch

  Using cached torch-2.9.0-cp312-cp312-win_amd64.whl.metadata (30 kB)
Using cached torch-2.9.0-cp312-cp312-win_amd64.whl (109.3 MB)
  Attempting uninstall: torch
    Found existing installation: torch 2.7.1+cu118
    Uninstalling torch-2.7.1+cu118:
      Successfully uninstalled torch-2.7.1+cu118


  You can safely remove it manually.
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
torchaudio 2.7.1+cu118 requires torch==2.7.1+cu118, but you have torch 2.9.0 which is incompatible.
torchvision 0.22.1+cu118 requires torch==2.7.1+cu118, but you have torch 2.9.0 which is incompatible.
